extensions-core/lookups-cached-single/src/main/java/io/druid/server/lookup/jdbc/JdbcDataFetcher.java

```diff
@@ -0,0 +1,230 @@
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ */
+
+package io.druid.server.lookup.jdbc;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.metamx.common.logger.Logger;
+import io.druid.metadata.MetadataStorageConnectorConfig;
+import io.druid.server.lookup.DataFetcher;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.TransactionCallback;
+import org.skife.jdbi.v2.TransactionStatus;
+import org.skife.jdbi.v2.tweak.HandleCallback;
+import org.skife.jdbi.v2.util.StringMapper;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+public class JdbcDataFetcher implements DataFetcher<String, String>
+{
+  private static final Logger LOGGER = new Logger(JdbcDataFetcher.class);
+  private static final int DEFAULT_STREAMING_FETCH_SIZE = 1000;
+
+  @JsonProperty
+  private final MetadataStorageConnectorConfig connectorConfig;
+  @JsonProperty
+  private final String table;
+  @JsonProperty
+  private final String keyColumn;
+  @JsonProperty
+  private final String valueColumn;
+  @JsonProperty
+  private final int streamingFetchSize;
+
+  private final String fetchAllQuery;
+  private final String fetchQuery;
+  private final String reverseFetchQuery;
+  private final DBI dbi;
+
+  public JdbcDataFetcher(
+      @JsonProperty("connectorConfig") MetadataStorageConnectorConfig connectorConfig,
+      @JsonProperty("table") String table,
+      @JsonProperty("keyColumn") String keyColumn,
+      @JsonProperty("valueColumn") String valueColumn,
+      @JsonProperty("streamingFetchSize") Integer streamingFetchSize
+  )
+  {
+    this.connectorConfig = Preconditions.checkNotNull(connectorConfig, "connectorConfig");
+    this.streamingFetchSize = streamingFetchSize == null ? DEFAULT_STREAMING_FETCH_SIZE : streamingFetchSize;
+    Preconditions.checkNotNull(connectorConfig.getConnectURI(), "connectorConfig.connectURI");
+    this.table = Preconditions.checkNotNull(table, "table");
+    this.keyColumn = Preconditions.checkNotNull(keyColumn, "keyColumn");
+    this.valueColumn = Preconditions.checkNotNull(valueColumn, "valueColumn");
+
+    this.fetchAllQuery = String.format(
+        "SELECT %s, %s FROM %s",
+        this.keyColumn,
+        this.valueColumn,
+        this.table
+    );
+    this.fetchQuery = String.format(
+        "SELECT %s FROM %s WHERE %s = :val",
+        this.valueColumn,
+        this.table,
+        this.keyColumn
+    );
+    this.reverseFetchQuery = String.format(
+        "SELECT %s FROM %s WHERE %s = :val",
+        this.keyColumn,
+        this.table,
+        this.valueColumn
+    );
+    dbi = new DBI(
```

Comment:
This will probably need some sort of thing to make sure the jdbc driver of interest is visible from the classloader of the extension. Since that is really really hard to unit test, I suggest giving it a try manually and reporting back. See https://github.com/druid-io/druid/issues/2345 for related issue.